2016-04-30  Nilesh Chander Gupta  <chander@lnxsrv01.seas.ucla.edu>

	* timespec2.h: Now correctly handles cases of overflow

--- ./lib/timespec.h	2005-09-21 23:05:39.000000000 -0700
+++ timespec2.h	2016-04-30 17:56:11.439877000 -0700
@@ -46,7 +46,14 @@
 timespec_cmp (struct timespec a, struct timespec b)
 {
   int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+  if (!(a.tv_sec < 0 && b.tv_sec < 0)){
+      if (a.tv_sec < 0)
+	return -1;
+      else if (b.tv_sec < 0)
+	return 1;
+    }
+  else  
+    return diff ? diff : a.tv_nsec - b.tv_nsec;
 }
 
 # if ! HAVE_DECL_NANOSLEEP
